controllers/kustomization_controller.go

```diff
@@ -768,6 +779,24 @@ func (r *KustomizationReconciler) event(kustomization kustomizev1.Kustomization,
 	}
 }
 
+func (r *KustomizationReconciler) recordReadiness(kustomization kustomizev1.Kustomization, deleted bool) {
+	if r.MetricsRecorder == nil {
+		return
+	}
+
+	objRef, err := reference.GetReference(r.Scheme, &kustomization)
+	if err != nil {
+		r.Log.WithValues(
+			strings.ToLower(kustomization.Kind),
+			fmt.Sprintf("%s/%s", kustomization.GetNamespace(), kustomization.GetName()),
+		).Error(err, "unable to record readiness metric")
+		return
+	}
+	if rc := meta.GetCondition(kustomization.Status.Conditions, meta.ReadyCondition); rc != nil {
```

Comment:
From the kubernetes [API conventions](https://github.com/evankanderson/k8s-community/blob/master/contributors/devel/sig-architecture/api-conventions.md):

> The absence of a condition should be interpreted the same as Unknown.

So even if the ready condition is not found yet shouldn't we still record the readiness as unknown?

Reply:
I don't see how this is possible with the current code but I've done as you asked.